src/rating.js

```diff
@@ -94,6 +94,11 @@ export default class Rating extends PureComponent {
     }
   }
 
+
+  componentWillReceiveProps(nextProps) {
+    this.animate(nextProps.initial, false)()
```

Comment:
calling setState here should be enough ?

Reply:
No, it should not be enough.
It works exactly how I implemented it.